### PR TITLE
Implement `parseBytesRef` for TimeSeriesRoutingHashFieldType

### LINKED ISSUE #112399: UnsupportedOperationException related to SearchAfterBuilder
### Elasticsearch Version

serverless

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Linux

### Problem Description

An error was found with the following stack trace:

```
Failed to execute phase [query], all shards failed; shardFailures {[nBwrVb73SgWRggIXMIPTKg][index-forensic-records][0]: org.elasticsearch.transport.RemoteTransportException: [es-es-search-7b47df8b76-6c7lh][100.65.128.14:9300][indices:data/read/search[phase/query]]
Caused by: java.lang.UnsupportedOperationException
	at org.elasticsearch.server@8.16.0/org.elasticsearch.search.DocValueFormat.parseBytesRef(DocValueFormat.java:83)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.search.searchafter.SearchAfterBuilder.convertValueFromSortType(SearchAfterBuilder.java:190)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.search.searchafter.SearchAfterBuilder.convertValueFromSortField(SearchAfterBuilder.java:149)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.search.searchafter.SearchAfterBuilder.buildFieldDoc(SearchAfterBuilder.java:117)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.search.SearchService.parseSource(SearchService.java:1443)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.search.SearchService.createContext(SearchService.java:1082)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:685)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.search.SearchService.lambda$executeQueryPhase$2(SearchService.java:559)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionRunnable$3.accept(ActionRunnable.java:78)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionRunnable$3.accept(ActionRunnable.java:75)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:100)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:991)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
}
```

### Steps to Reproduce

Unknown at this time.

### Logs (if relevant)

Unknown at this time.

### PR DESCRIPTION
**Pull request by andreidan**
This implements the `parseBytesRef` method for the `_ts_routing_hash` field so we can parse the values generated by the companion `format` method. We parse the values when fetching them from the source when the field is used as a `sort` paired with `search_after`.

Before this change a sort by and search_after `_ts_routing_hash` would yield an `UnsupportedOperationException`

Fixes #112399

### CODE REVIEW

**Comment by andreidan** [comment]
cc @javanna as we've talked about this

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/25_id_generation.yml [hunk 1]
```diff
@@ -152,6 +152,55 @@ generates a consistent id:
   - match: { hits.hits.8._source.@timestamp: 2021-04-28T18:52:04.467Z }
   - match: { hits.hits.8._source.k8s.pod.uid: 947e4ced-1786-4e53-9e0c-5c447e959507 }
 
+  - do:
```
Comment (andreidan) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/25_id_generation.yml:155]: This repeats the previous search in the test (to keep the test setup) but
adds the necessary conditions (sorting and search_after) in order
to exercise the modified code.

Comment (andreidan) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/25_id_generation.yml:155]: Oops, fix incoming to add the necessary version skips

**Review by javanna — COMMENTED** [review]
Good catch! I will let storage engine folks review this. I think a unit test around the fix would not hurt also.

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks Andrei!

**Pull request merged**